test/conftest.py

```diff
@@ -734,3 +777,157 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
+
+
+@pytest.fixture(scope="session")
+def differential_expression_matrix_no_genome_set(
+    diff_exp_matrix_no_genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(
+        DIFFERENTIAL_EXPRESSION_MATRIX,
+        set_api_client,
+        context,
+        diff_exp_matrix_no_genome_refs,
+        ws_id,
+        "no_genome",
+    )
+
+
+@pytest.fixture(scope="session")
+def expression_set(
+    expression_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(EXPRESSION, set_api_client, context, expression_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def feature_set_set(
+    feature_set_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(FEATURE_SET, set_api_client, context, feature_set_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_feature_set_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(FEATURE_SET, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def genome_set(
+    genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(GENOME, set_api_client, context, genome_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_genome_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(GENOME, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def kbase_search_genome_set(
+    genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list],
+    ws_id: int,
+) -> dict[str, Any]:
+    """Set of KBaseSearch genomes."""
+    return save_kbase_search_set(set_api_client, context, genome_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def reads_set(
+    reads_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(READS, set_api_client, context, reads_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def empty_reads_set(
+    set_api_client: SetAPI, context: dict[str, str | list[Any]], ws_id: int
+) -> dict[str, Any]:
+    return save_empty_set(READS, set_api_client, context, ws_id)
+
+
+@pytest.fixture(scope="session")
+def reads_alignment_set(
+    alignment_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, str | list[Any]],
+    ws_id: int,
+) -> dict[str, Any]:
+    return save_set(READS_ALIGNMENT, set_api_client, context, alignment_refs, ws_id)
+
+
+@pytest.fixture(scope="session")
+def random_fixture(request: FixtureRequest) -> dict[str, Any]:
+    """Pick a random fixture to use for testing."""
```

Comment:
this is pretty cool (I asked ChatGPT how to do... if only I were that clever myself!)